open-api/rest-catalog-open-api.yaml

```diff
@@ -541,6 +541,216 @@ paths:
+        such as selected columns, filters, snapshot range, case sensitivity, etc.
+        In the event that the plan tasks or file scan tasks are not ready to be served, the service will return a `plan-id`,
+        which can be used as input in the `GetTasksStatus`.
+        
+        Requires that the client specifies only a `snapshot-id` for a snapshot scan, or for performing incremental scans only provide 
+        a `start-snapshot-id` and an `end-snapshot-id`. 
+      
+        Each `PlanTask`, can be used as input in the `RetrieveTasks` API
+        to request a subset of file scan tasks in a table scan.
+        This mechanism allows clients to distribute and parallelize the entire table scan planning process.
+      operationId: planTable
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PlanTableRequest'
+      responses:
+        200:
+          $ref: '#/components/responses/PlanTableResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        403:
+          $ref: '#/components/responses/ForbiddenResponse'
+        404:
+          description:
+            Not Found
+            - NoSuchTableException, the table does not exist
+            - NoSuchNamespaceException, the namespace does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergErrorResponse'
+              examples:
+                TableDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        406:
+          $ref: '#/components/responses/UnsupportedOperationResponse'
+        419:
+          $ref: '#/components/responses/AuthenticationTimeoutResponse'
+        503:
+          $ref: '#/components/responses/ServiceUnavailableResponse'
+        5XX:
+          $ref: '#/components/responses/ServerErrorResponse'
+
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/plan/{id}:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+      - $ref: '#/components/parameters/id'
+
+    get:
+      tags:
+        - Catalog API
+      summary: Uses a `plan-id` as input to get status of `planTable`. Returns a list of `PlanTask`, a list of `FileScanTask`, or both when plan is complete.
+      operationId: GetTaskStatus
+      description:
+        Gets the status of a plan by using a `plan-id` which can be obtained from `planTable`.
+        If the plan is not completed, returns a `plan-status` representing the state of the plan.
+        If the plan is completed, returns a list of `PlanTask` or `FileScanTask`.
+      
```

Comment:
Nit: Don't think you need a newline here

Reply:
will fix

Reply:
@amogh-jahagirdar, I think this newline is needed. These blocks are not formatted how you might expect so you need a blank newline to start a new paragraph.

Reply:
I tested this and _two_ newlines are needed for the Swagger editor.